includes/widgets/video.php

```diff
@@ -220,30 +420,65 @@ protected function render( $instance = [] ) {
 	}
 
 	public function filter_oembed_result( $html, $url, $args ) {
-		$youtube_params = [];
-
-		if ( 'yes' === $this->_current_instance['yt_autoplay'] )
-			$youtube_params[] = 'autoplay=1';
-
-		if ( 'no' === $this->_current_instance['yt_rel_videos'] )
-			$youtube_params[] = 'rel=0';
+		$params = [];
+
+		if ( 'youtube' === $this->_current_instance['video_type'] ) {
+			foreach ( $this->_current_instance as $option => $values ) {
+				if ( false !== strpos( $option, 'yt' ) ) {
+					$key = str_replace( 'yt_', '', $option );
+					$value = ( 'yes' === $this->_current_instance[ $option ] ) ? '1' : '0';
+					$params[ $key ] = $value;
+				}
+			}
+
+			$params['wmode'] = 'opaque';
+		}
 
-		if ( 'no' === $this->_current_instance['yt_controls'] )
-			$youtube_params[] = 'controls=0';
+		if ( 'vimeo' === $this->_current_instance['video_type'] ) {
+			foreach ( $this->_current_instance as $option => $values ) {
+				if ( false !== strpos( $option, 'vimeo' ) ) {
+					$key = str_replace( 'vimeo_', '', $option );
+					$value = ( 'yes' === $this->_current_instance[ $option ] ) ? '1' : '0';
+					$params[ $key ] = $value;
+				}
+			}
 
-		if ( 'no' === $this->_current_instance['yt_showinfo'] )
-			$youtube_params[] = 'showinfo=0';
+			$params['color'] = str_replace( '#', '', $this->_current_instance['vimeo_color'] );
+		}
 
-		// TODO: Check if is youtube link
-		$youtube_params[] = 'wmode=opaque';
+		if ( ! empty( $params ) ) {
+			preg_match( '/<iframe.*src=\"(.*)\".*><\/iframe>/isU', $html, $matches );
+			$url = esc_url( add_query_arg( $params, $matches[1] ) );
 
-		if ( ! empty( $youtube_params ) ) {
-			$separator = '&amp;';
-			$html = str_replace( '?feature=oembed', '?feature=oembed' . $separator . implode( $separator, $youtube_params ), $html );
+			$html = str_replace( $matches[1], $url, $html );
 		}
 
 		return $html;
 	}
 
+	public function build_hosted_params() {
```

Comment:
Not Good !
1) If the `$this->_current_instance['video_type']` is not 'hosted', quite from the method.
2) This method need to return always value.
3) Change the name to `get_hosted_params()`.